velox/functions/prestosql/registration/ArrayFunctionsRegistration.cpp

```diff
@@ -135,7 +136,11 @@ void registerArrayFunctions(const std::string& prefix) {
   VELOX_REGISTER_VECTOR_FUNCTION(udf_zip, prefix + "zip");
   VELOX_REGISTER_VECTOR_FUNCTION(udf_zip_with, prefix + "zip_with");
   VELOX_REGISTER_VECTOR_FUNCTION(udf_array_position, prefix + "array_position");
-  VELOX_REGISTER_VECTOR_FUNCTION(udf_array_shuffle, prefix + "shuffle");
+  exec::registerStatefulVectorFunction(
```

Comment:
It is stateless

Reply:
Thanks, it should be stateful after this change, the state should be `randGen_`, we want it be assigned a new  random seed  for every expr compilation.